TKGame/BackEnd/EntityManager.cs

```diff
@@ -89,7 +90,33 @@ public static void Update(GameTime gameTime)
         public static void Draw(SpriteBatch spriteBatch)
         {
             foreach (var entity in entities)
+            {
                 entity.Draw(spriteBatch);
+                DrawHealth(spriteBatch, entity);
+            }
+        }
+
+        public static void DrawHealth(SpriteBatch spriteBatch, Entity entity)
+        {
+            if (entity.needsHealth)
+            {
+                entity.healthTexture = new Texture2D(spriteBatch.GraphicsDevice, 1, 1);
+                entity.healthBar = new Rectangle((int)entity.Position.X - 50, (int)entity.Position.Y - 70, 100 * (entity.health / entity.originalHealth), 10);
+                entity.healthTexture.SetData(new Color[] { Color.Red });
+                spriteBatch.Draw(entity.healthTexture, entity.healthBar, Color.White);
+            }
+        }
+
+        public static void DamageEnemy()
+        {
+            foreach (Entity entity in entities)
+            {
+                if (entities[0].hitBox.Intersects(entity.hitBox) && entity != entities[0])
```

Comment:
How come entities[0] is specified here? Are you trying to access the player? Player.Instance for that or query the list for the player instead so this doesn't potentially try to call on an entity other than what is intended